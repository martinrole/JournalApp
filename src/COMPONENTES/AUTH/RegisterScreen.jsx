import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../HOOKS/useForm'
import validator from 'validator'
import { useDispatch } from 'react-redux'
import { actionRegisterAsync } from '../../ACTIONS/actionAuth'

const RegisterScreen = () => {

    const [formError, setFormError] = useState(null)
    const dispatch = useDispatch()

    const [ formValues, handleInputChange ] = useForm({
        nombre: 'Martin',
        correo: 'pruebaMartin@hotmail.com',
        contrasena: '12nj3nh4',
        contrasena2: '12nj3nh4'
    })

    const { nombre, correo, contrasena, contrasena2 } = formValues

    const handleRegister = (e) => {
        e.preventDefault()
        
        if ( isFormValid() ) {
            // console.log("Form válido")
            dispatch( actionRegisterAsync( nombre, correo, contrasena ) )
        }
    }

    const isFormValid = () => {

        if( !nombre.trim() ) {
            setFormError("Ingrese un nombre")
            return false
        } else if ( !validator.isEmail( correo) ) {
            setFormError("Correo no es válido")
            return false
        } else if ( contrasena < 5 ) {
            setFormError("Contraseña debe tener al menos 6 carácteres")
            return false
        } else if ( contrasena !== contrasena2 ) {
            setFormError("Contraseñas ingresadas no concuerdan")
            return false
        }

        setFormError(null)
        return true
    }

    return (
        <>
            <h3 className="auth__title">Register Screen</h3>
            
            <form onSubmit={ handleRegister }>
                
                {
                    formError && <div className="auth__alert-error"> { formError } </div>
                }
                
                <input 
                    type="text" 
                    placeholder="Nombre" 
                    name="nombre" 
                    className="auth__input" 
                    autoComplete="off"
                    value={ nombre }
                    onChange={ handleInputChange }
                />
                <input 
                    type="text" 
                    placeholder="Correo" 
                    name="correo" 
                    className="auth__input" 
                    autoComplete="off"
                    value={ correo }
                    onChange={ handleInputChange }

                />
                <input 
                    type="password" 
                    placeholder="Contraseña" 
                    name="contrasena" 
                    className="auth__input"
                    value={ contrasena }
                    onChange={ handleInputChange }

                />
                <input 
                    type="password" 
                    placeholder="Confirmar contraseña" 
                    name="contrasena2" 
                    className="auth__input"
                    value={ contrasena2 }
                    onChange={ handleInputChange }

                />

                <button type="submit" className="btn btn-primario btn-bloque mt-1">Register</button>
            
            </form>
           
            <Link to="/auth/login" className="auth__new-count"> Already registered?</Link>

        </>
    )
}

export default RegisterScreen