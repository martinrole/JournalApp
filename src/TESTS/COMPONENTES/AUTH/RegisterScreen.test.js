import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';
import '@testing-library/jest-dom'
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import { MemoryRouter } from 'react-router-dom';

import RegisterScreen from '../../../COMPONENTES/AUTH/RegisterScreen';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        cargando: false
    }
}

let store = mockStore( initState )

const wrapper = mount( 
    <Provider store={ store }>
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
)

describe('Pruebas en componente RegisterScreen', () => {

    beforeEach(() => {
        //Limpiar el store despuès de cada test
        store = mockStore( initState )
        jest.clearAllMocks()
    })

    test('Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe mostrar la caja de alerta con error del form', () => {

        const inputCorreo = wrapper.find('input[name="correo"]')
        inputCorreo.simulate('change',{
            target:{
                value:'',
                name:'correo'
            }
        })

        wrapper.find('form').simulate('submit')

        const msgError = wrapper.find('.auth__alert-error')

        expect( msgError.exists() ).toBe(true)
        expect( msgError.text().trim() ).toBe('Correo no es válido')

    })
})