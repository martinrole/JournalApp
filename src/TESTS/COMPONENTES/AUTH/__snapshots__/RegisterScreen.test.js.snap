// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente RegisterScreen Debe mostrar el componente correctamente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register Screen
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="nombre"
      onChange={[Function]}
      placeholder="Nombre"
      type="text"
      value="Martin"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="correo"
      onChange={[Function]}
      placeholder="Correo"
      type="text"
      value="pruebaMartin@hotmail.com"
    />
    <input
      className="auth__input"
      name="contrasena"
      onChange={[Function]}
      placeholder="Contraseña"
      type="password"
      value="12nj3nh4"
    />
    <input
      className="auth__input"
      name="contrasena2"
      onChange={[Function]}
      placeholder="Confirmar contraseña"
      type="password"
      value="12nj3nh4"
    />
    <button
      className="btn btn-primario btn-bloque mt-1"
      type="submit"
    >
      Register
    </button>
  </form>,
  <a
    className="auth__new-count"
    href="/auth/login"
    onClick={[Function]}
  >
     Already registered?
  </a>,
]
`;
